# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/spellchecker/TextServicesDocument.h
# Commit: da317e009565
# Full Hash: da317e0095654ccecf40fec345180dfd4aa4001c
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-07-13 21:42:32
# Regressor Bug: 1718924
# File Overlap Count: 2
# Description:
#   Bug 1718924 - part 16: Move `TextServicesDocument::RemoveInvalidOffsetEntries()` to `OffsetEntry` r=m_kato
#   
#   Depends on D119162
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119163
# ==============================================================================

diff -r b771c6d7f21a -r da317e009565 editor/spellchecker/TextServicesDocument.h
--- a/editor/spellchecker/TextServicesDocument.h	Tue Jul 13 10:57:48 2021 +0000
+++ b/editor/spellchecker/TextServicesDocument.h	Tue Jul 13 11:03:56 2021 +0000
@@ -89,6 +89,11 @@
      */
     nsresult SplitElementAt(size_t aIndex, uint32_t aOffsetInTextNode);
 
+    /**
+     * Remove all `OffsetEntry` elements whose `mIsValid` is set to false.
+     */
+    void RemoveInvalidElements();
+
     class Selection final {
      public:
       size_t StartIndex() const {
@@ -367,8 +372,6 @@
 
   bool SelectionIsCollapsed() const;
   bool SelectionIsValid() const;
-
-  nsresult RemoveInvalidOffsetEntries();
 };
 
 }  // namespace mozilla
